File: codes/jiheon/933_numberofrecentcalls.java
class RecentCounter {

    private final Deque<Integer> q = new ArrayDeque<>();

    public RecentCounter() {

    }

    public int ping(int t) {
        while (q.peekFirst() != null && q.peekFirst() < t - 3000) {
            q.pollFirst();
        }
        q.addLast(t);
        return q.size();
    }
}

File: codes/jiheon/933_numberofrecentcalls.kt
class RecentCounter() {

    private val q = ArrayDeque<Int>()

    fun ping(t: Int): Int {
        while (q.firstOrNull()?.let { it < t - 3000 } == true) {
            q.removeFirst()
        }
        q.add(t)
        return q.size
    }

}